Cleanups in connectors logging

dmatej opened the pull request:
- some messages were broken
- some causes were lost
- several logs were redundant
- fixed some javadocs
- these changes were done locally when I was working on #24335 and #24337 
- 08d7fc5493a81505ab9b307aa42b72158333a3fb is rather trivial, I wasn't sure if it cannot lead to negative pool size or size higher than max. Now it is covered by a unit test.

dmatej committed:
Cleanups in connectors logging

- some messages were broken
- some causes were lost
- several logs were redundant
- fixed some javadocs

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

hs536 commented on appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/allocator/LocalTxConnectorAllocator.java:
```diff
@@ -81,17 +80,6 @@ public ResourceHandle createResource()
             return resource;
         } catch (ResourceException ex) {
-            Object[] params = new Object[]{spec.getPoolInfo(), ex.toString()};
-            _logger.log(Level.WARNING, "poolmgr.create_resource_error", params);
```
Is the deletion of this log as intended? (delegate log to the caller who catches the PoolingException?

dmatej replied:
Yes, because the exception is wrapped and thrown again several times. In effect when something broke, I had 5 times the same error in the server.log file, yet once with a broken message.

But based on your comment it seems to me that some paths may result in hiding the exception, so for sure I will revisit all paths and ensure that the exception is logged exactly once including stacktrace.

dmatej replied:
TODO for next commits of this PR: catch and process the exception, don't repeat logging and rethrowing.

dmatej replied:
I have revisited that one more. It is finally called just from one place in ConnectionPool, which throws the exception including the cause OR sleeps and gives it another try. Now, before the sleep, it should log it as a warning including some message what will happen next.

dmatej commented on appserver/connectors/connectors-runtime/src/main/java/com/sun/enterprise/resource/allocator/ConnectorAllocator.java:
```diff
@@ -143,30 +140,14 @@ public ConnectorAllocator(PoolManager poolMgr,
-                _logger.log(Level.FINE,"Resource Exception while creating resource",ex);
-            }
-
-            if (ex.getLinkedException() != null) {
```
This is obsoleted deprecated getter, generally not supported any more, and also ignored on other places. 
TODO in next commits of this PR: Remove all calls of the setter.

dmatej committed:
Usages of setLinkedException replaced by standard "cause"

- obsoleted and deprecated code

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Fixed race conditions in UnpooledResource

- new PoolSize class with atomic operations + unit test
- guaranteed that we cannot go under zero
- guaranteed that we will not provide more than maxPoolSize

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Yet one iteration of logging cleanup in connectors

- createResource is called just from ConnectionPool where is the exception
  processed too. Warning will be logged for each failure there, until retries
  exceed configured count of attempts. Then instead of warning it will throw
  an exception.
- Removed also i18n properties, fixed messages, unused dumpPoolStatus

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Fixed task count in PoolSizeTest

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Fixed unresolved log messages, dead code

- this package should get more attention

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Fixed PoolSize and unit test

- Funny trap I made for myself.
- Integration tests detected the bug (jdbc_all)

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Fixed NPE in jdbc_all tests

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>
